### PR TITLE
storage: relax TestAcquireLeaderLease requirements

### PR DESCRIPTION
**Pull request by tamird**
Some random logging improvements here, too.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3754)

### CODE REVIEW

File: util/log/log.go [hunk 1]
```diff
@@ -86,8 +86,8 @@ func Infof(format string, args ...interface{}) {
 
 // InfoDepth logs to the INFO log, offsetting the caller's stack frame by
 // 'depth'.
-func InfoDepth(depth int, args ...interface{}) {
-	logDepth(nil, depth+1, InfoLog, "", args)
+func InfoDepth(depth int, format string, args ...interface{}) {
```
Comment (bdarnell) [inline util/log/log.go]: The lack of a `format` argument to these functions was deliberate: they are `Depth` variants of `log.Info`, not `log.Infof`.  If they take a format they should be renamed to `log.InfofDepth`.

The fact that our logging implementation supports an empty format string as equivalent to the non-`f` version of the logging function is slightly surprising (and inconsistent with the functions in package `fmt`), and either needs to be documented or we need to keep non-`f` versions of these logging functions for use in the raft log adapter.

File: storage/raft.go [hunk 2]
```diff
@@ -48,68 +48,59 @@ type raftLogger struct {
 	group uint64
 }
 
-func (r *raftLogger) prependContext(format string, v []interface{}) string {
-	var s string
+func (r *raftLogger) context() string {
```
Comment (bdarnell) [inline storage/raft.go]: Add a comment that bad things will happen if this method returns a string that may contain `%` characters.

File: storage/raft.go [hunk 3]
```diff
@@ -48,68 +48,59 @@ type raftLogger struct {
...
-		s := r.prependContext(format, v)
-		log.InfoDepth(1, s)
+		log.InfoDepth(1, r.context()+format, v...)
 	}
 }
 
 func (*raftLogger) Info(v ...interface{}) {
 	if log.V(1) {
-		log.InfoDepth(1, v...)
+		log.InfoDepth(1, "", v...)
```
Comment (bdarnell) [inline storage/raft.go]: Unrelated to this change, but the non-`f` logging functions should really prepend `context()` to their arguments. (I'm not sure if these are actually used). Maybe leave a TODO.

**Comment by bdarnell** [comment]
LGTM

File: storage/queue.go [hunk 4]
```diff
@@ -114,7 +114,7 @@ type queueLog struct {
 
 func (l queueLog) Infof(logv bool, format string, a ...interface{}) {
 	if logv {
-		log.Infof(l.prefix+format, a...)
+		log.InfoDepth(1, l.prefix+format, a...)
```
Comment (petermattis) [inline storage/queue.go]: According to Ben, this is not equivalent. You'd need to add an `InfofDepth` method.

Comment (tamird) [inline storage/queue.go]: Why is this not equivalent? I don't follow.

File: storage/raft.go (hunk 3 above)
Comment (tamird) [inline storage/raft.go]: easy enough to just fix this, done.

File: util/log/log.go (hunk 1 above)
Comment (tamird) [inline util/log/log.go]: agreed; i've documented this oddity.

File: storage/raft.go (hunk 2 above)
Comment (tamird) [inline storage/raft.go]: done

File: storage/queue.go (hunk 4 above)
Comment (petermattis) [inline storage/queue.go]: Ah, you've changed it to `InfofDepth`. LG.

**Pull request merged**